### ISSUE TITLE
Optimize Mip Generation

### ISSUE THREAD

**Issue by bjjones** [JiangYizhou/aquarium-test#4]
Aquarium currently suffers from a very long startup time. The largest contributor to this is mipmap generation for all textures. Currently this occurs single threaded on the CPU. The ideal solution would be to upload each full texture to the GPU and use a compute shader to generate mipmaps.

**Comment by JiangYizhou** [comment]
Yes, I agree generate mipmaps by compute shader should be most efficient but image isn't supported in compute shader in dawn yet. Shall we put this in lower priority?  @Kangz @bjjones @Jiawei-Shao

**Comment by Kangz** [comment]
I think it is fine to delay, there's two options to generate mipmaps efficiently on the GPU:

 - Using a chain of render pass drawing the previous level using linear filtering in the lower level. This requires the "fine-grained resource tracking" Dawn feature.
 - Using compute shaders, this requires storage textures.

**Comment by bjjones** [comment]
I see. It's fine to put this as lower priority - this is just to make the workload a bit nicer to use. Issues in the main render loop are higher priority.
